### DISCUSSION TITLE
Why is the underscore required for many-to-many implicit?

### DISCUSSION THREAD

**Discussion by edeesis** [prisma/prisma#12210] [Q&A] [answered]
We're beginning a migration from TypeORM to Prisma, and the [implicit many-to-many](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations) caught my eye.

However, since our tables are already created and named in the format of primaries_secondaries (Rails-y way of naming join tables), we can't rename the tables, but Prisma requires them to start with an underscore.

Why is this? Why can't we just name the tables however we want using the `@map`

**Comment by nikolasburk** [comment, marked as answer]
Hey @edeesis, the design of the relation tables is a legacy from the the Prisma 1 days, so the requirements are there mainly for historic reasons at the moment. We are however planning to improve the situation and make m-n relations more flexible in the future, since that will most likely be a breaking change it's a bit more work though and will only be introduced with a future _major_ version update.

**Reply by edeesis** [reply]
Thanks @nikolasburk for your answer. All that makes sense, and while it's a shame we have to wait til Prisma 4 to get the nicer API, I'm sure it'll be worth the wait.

**Reply by nikolasburk** [reply]
I just checked back with Engineering and as of now it's unclear when exactly that redesign is going to happen, so I've adjusted the message to only mention that it'll happen in a _major_ version release.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
